### DISCUSSION TITLE
Doom Emacs

### DISCUSSION THREAD

**Discussion by jespada** [dustinlyons/nixos-config#44] [Q&A] [answered]
Hi @dustinlyons ,
Thanks for your work, I'm moving all my nix setup to this model, I was wondering where or how would you go about adding Doom Emacs in this type of setup, I like the ideas behind https://github.com/MatthiasBenaets/nixos-config/blob/master/modules/editors/emacs/doom-emacs/default.nix.
Any comments would be helpful. 
Thanks

**Comment by dustinlyons** [comment, marked as answer]
Hi,

I agree that https://github.com/MatthiasBenaets/nixos-config/blob/master/modules/editors/emacs/doom-emacs/default.nix looks promising, especially given the broken state of `nix-doom-emacs`. Given it's a system option, you should be able to add something like his activationScript to `hosts/darwin/default.nix`.

You could also look to just add an overlay that does these steps, which runs on every build. Ultimately you want to test a series of steps (i.e a script) that bring Doom to the state you like, and then just stick those in either an activationScript or an overlay.

I personally like using overlays and this project runs each overlay as part of rebuild. This is a good reference to see how overlays work: https://github.com/jwiegley/nix-config/tree/master/overlays

**Reply by jespada** [reply]
@dustinlyons is there a reason why only `emacsclient` is on the $PATH and not `emacs` binary ? is this default in the overlay ?

**Reply by dustinlyons** [reply]
My configuration uses Emacs daemon mode, so emacsclient connects to the daemon. You can just as easily run the emacs binary but it will boot much slower. I prefer to always have it running, and my configuration is organized around that.

**Reply by jespada** [reply]
Ah make sense, that's what I've thought, doom emacs wants the `emacs` bin to be in the path at install time , but other than that I also like to run it as a daemon. 
If I want to have `emacs` in my path along with `emacsclient` where should I enable that? sorry for the newb question.

**Reply by dustinlyons** [reply]
The emacs binary sits in `/run/current-system/sw/bin`. You can add that to your path here: https://github.com/dustinlyons/nixos-config/blob/e07ef74ed0bcd99789f8bc4721a45d4db42b0d10/modules/shared/home-manager.nix#L41

**Reply by jespada** [reply]
Great, thanks again! [☕](https://www.buymeacoffee.com/dustinlyons1/c/8161433)
